File: src/anylayout/extras/LayoutContextUtility.java
package anylayout.extras;

import anylayout.LayoutContext;
import fj.F;

import java.awt.Component;

public class LayoutContextUtility
{
	public static final F<LayoutContext, Integer> getParentSize=new F<LayoutContext,Integer>()
	{
		public Integer f(final LayoutContext layoutContext)
		{
			return layoutContext.getParentSize();
		}
	};
	
	public static final F<LayoutContext, Integer> getPreferredSize=new F<LayoutContext,Integer>()
	{
		public Integer f(final LayoutContext layoutContext)
		{
			return layoutContext.getPreferredSize();
		}
	};

	public static F<LayoutContext,Integer> getFarOffset(final Component component)
	{
		return new F<LayoutContext,Integer>()
		{
			public Integer f(final LayoutContext context)
			{
				return context.getLayoutInfo(component).getFarOffset();
			}
		};
	}
}

File: src/anylayout/extras/LayoutContextUtility.scala
package anylayout.extras

import anylayout.LayoutContext

import java.awt.Component

object LayoutContextUtility {
 val getParentSize: LayoutContext => Int = _.parentSize
 val getPreferredSize: LayoutContext => Int = _.preferredSize
 def getFarOffset(component: Component): LayoutContext => Int = _.layoutInfoFor(component).getFarOffset
}
